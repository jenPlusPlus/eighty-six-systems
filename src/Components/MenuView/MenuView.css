.table-info-menuview {
  width: 120px;
  height: 40px;
  padding: 11px;
  text-align: center; }

.go-back {
  display: inline-block;
  margin: auto;
  padding: 5px; }

.table-button {
  background-color: #EFEFF0;
  border-radius: 20px; }
