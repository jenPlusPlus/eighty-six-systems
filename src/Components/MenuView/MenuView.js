import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

class MenuView extends Component {
  constructor() {
    super();
  }
  render() {
    return (
      <div className='menu-view'>
        <div className='go-back'>
          <Link to={`/${this.props.currentUser.loginCode}/tables`}>
            <h3 className='all-tables-button'>All Tables</h3>
          </Link>
          <Link to={`/${this.props.currentUser.loginCode}/tables/${this.props.currentTable.tableNumber}`}>
            <h3 className='table-info-menuview table-button'>Table {this.props.currentTable.tableNumber}</h3>
          </Link>
          <h3 className='table-info-menuview seat-button'>Seat {this.props.currentSeat.seatNumber}</h3>
          </div>
        <div className='menu'>
          <h3>MENU VIEW</h3>
        </div>
      </div>
    );
  }
}

MenuView.propTypes = {
  currentUser: PropTypes.object,
  currentTable: PropTypes.object,
  currentSeat: PropTypes.object
};

export default MenuView;
