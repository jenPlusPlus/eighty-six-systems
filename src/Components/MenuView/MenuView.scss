

.table-info-menuview {
  width: 120px;
  height: 40px;
  margin: auto;
  padding: 11px;
  margin-top: 15px;
  text-align: center;
  font-size: 19px;
}

.go-back {
  display: inline-block;
  margin: auto;
  padding: 5px;
}

.table-button {
  background-color: #EFEFF0;
  color: #33202A;
  width: 100px;
  height: 30px;
  padding: 8px;
  text-align: center;
  border-radius: 20px;
  font-size: 16px;
}
