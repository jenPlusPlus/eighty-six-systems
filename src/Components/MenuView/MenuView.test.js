import { shallow } from 'enzyme';
import MenuView from './MenuView';
import ReactDOM from 'react-dom';
import React from 'react';

describe('MenuView snapshot', () => {

  it('should always match the snapshot', () => {
    const wrapper = shallow(<MenuView
      currentUser={{name: 'Jen', KitchenViewCode: '0000'}}
      currentTable={{tableNumber: '22'}}
      currentSeat={{
        tableNumber: '22',
        seatNumber: '1',
        order: [
        {item: 'Lamb', price: 29},
        {item: 'Burger', price: 19},
        {item:'Mahi Mahi', price: 27}
        ]
      }}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
