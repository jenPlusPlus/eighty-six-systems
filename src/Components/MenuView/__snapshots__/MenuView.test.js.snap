// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuView snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuView
    currentSeat={
        Object {
            "order": Array [
              Object {
                "item": "Lamb",
                "price": 29,
              },
              Object {
                "item": "Burger",
                "price": 19,
              },
              Object {
                "item": "Mahi Mahi",
                "price": 27,
              },
            ],
            "seatNumber": "1",
            "tableNumber": "22",
          }
    }
    currentTable={
        Object {
            "tableNumber": "22",
          }
    }
    currentUser={
        Object {
            "KitchenViewCode": "0000",
            "name": "Jen",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <Redirect
          push={false}
          to="/login"
/>,
      ],
      "className": "menu-view-wrapper",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "push": false,
          "to": "/login",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <Redirect
            push={false}
            to="/login"
/>,
        ],
        "className": "menu-view-wrapper",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "push": false,
            "to": "/login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
